run.sh

```diff
@@ -2,16 +2,29 @@
 
 LATEST_HUGO_VERSION=0.14
 
-# check if curl is installed
-# Install otherwise
-if [ "$(which curl)" == "" ]; then
-    if [ "$(which apt-get)" != "" ]; then
-        apt-get update
-        apt-get install -y curl
-    else
-        yum install -y curl
+command_exists()
+{
+    hash "$1" 2>/dev/null
+}
+
+install_hugo()
+{
+    # check if curl is installed
+    # install otherwise
+    if ! command_exists curl; then
+        if command_exists apt-get; then
+            apt-get update
+            apt-get install -y curl
+        else
+            yum install -y curl
+        fi
     fi
-fi
+    
+    cd $WERCKER_STEP_ROOT    
+    curl -sL https://github.com/spf13/hugo/releases/download/v${WERCKER_HUGO_BUILD_VERSION}/hugo_${WERCKER_HUGO_BUILD_VERSION}_linux_amd64.tar.gz -o ${WERCKER_STEP_ROOT}/hugo_${WERCKER_HUGO_BUILD_VERSION}_linux_amd64.tar.gz
+    tar xzf hugo_${WERCKER_HUGO_BUILD_VERSION}_linux_amd64.tar.gz
+    export HUGO_COMMAND=${WERCKER_STEP_ROOT}/hugo_${WERCKER_HUGO_BUILD_VERSION}_linux_amd64/hugo_${WERCKER_HUGO_BUILD_VERSION}_linux_amd64
```

Comment:
After thinking about it yes, I would prefer them to be regular variables. Mostly because I don't think it's a good idea to have other steps potentially depending on variables set in this step. As it's technically possible for subsequent steps to run on different docker containers you can't be 100% certain that they'll be available.